src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -144,4 +115,46 @@ void Series::fillDefaultSeriesIfEmpty()
     fillIfEmpty(upperRuleCurve, 1.0);
 }
 
+bool Series::validate() const
+{
+    if (maxInjectionModulation.size() != HOURS_PER_YEAR
+        || maxWithdrawalModulation.size() != HOURS_PER_YEAR || inflows.size() != HOURS_PER_YEAR
+        || lowerRuleCurve.size() != HOURS_PER_YEAR || upperRuleCurve.size() != HOURS_PER_YEAR)
+    {
+        logs.warning() << "Size of series for short term storage is wrong";
+        return false;
+    }
+
+    auto checkVectBetween = [](const std::vector<double>& v) {
+        return std::all_of(v.begin(), v.end(), [](double d){ return (d >= 0.0 && d <= 1.0); });
+    };
+
+    if (!checkVectBetween(maxInjectionModulation) || !checkVectBetween(maxWithdrawalModulation))
+    {
+        logs.warning() << "Values for PMAX series should be between 0 and 1";
+        return false;
+    }
+
+    if (!checkVectBetween(lowerRuleCurve) || !checkVectBetween(upperRuleCurve))
+    {
+        logs.warning() << "Values for rule curve series should be between 0 and 1";
+        return false;
+    }
+
+    return true;
+}
+
+bool Series::validateRuleCurve() const
```

Comment:
As you see, there are already some checks on rule curves done in **Series::validate()**.
And now we make new checks on rule curves in another function **Series::validateRuleCurve()**.
What I recommend : 
Keep a public **Series::validate()** member function and add a new private validate functions per kind of  time series
```C++
bool Series::validate() const
{
    // General
    if (maxInjectionModulation.size() != HOURS_PER_YEAR  ||  ....)
    {
        ...
    } 
    if (! validModulationTimeSeries())
        return false;
    if (! validRulesCurvesTimeSeries())
        return false;
    ...
    // Or we could also have : 
    return validModulationTimeSeries() &&  
           validRulesCurvesTimeSeries();
}
```